MaterialDesignThemes.Wpf/Stepper.cs

```diff
@@ -0,0 +1,345 @@
+            CommandBindings.Add(new CommandBinding(CancelCommand, CancelHandler, CanExecuteCancel));
+            CommandBindings.Add(new CommandBinding(ContinueCommand, ContinueHandler, CanExecuteContinue));
+            CommandBindings.Add(new CommandBinding(StepSelectedCommand, StepSelectedHandler, CanExecuteStepSelectedHandler));
+        }
+
+        private void LoadedHandler(object sender, RoutedEventArgs args)
+        {
+            _controller.PropertyChanged += PropertyChangedHandler;
+
+            // there is no event raised if the Content of a ContentControl changes
+            //     therefore trigger the animation in code
+            PlayHorizontalContentAnimation();
+        }
+
+        private void UnloadedHandler(object sender, RoutedEventArgs args)
+        {
+            _controller.PropertyChanged -= PropertyChangedHandler;
+        }
+
+        private bool ValidateActiveStep()
+        {
+            IStep step = _controller.ActiveStepViewModel?.Step;
+
+            if (step != null)
+            {
+                // call the validation method on the step itself
+                step.Validate();
+
+                // raise the event and call the command
+                StepValidationEventArgs eventArgs = new StepValidationEventArgs(StepValidationEvent, this, step);
+                RaiseEvent(eventArgs);
+
+                if (StepValidationCommand != null && StepValidationCommand.CanExecute(step))
+                {
+                    StepValidationCommand.Execute(step);
+                }
+
+                // the event handlers can set the validation state on the step
+                return !step.HasValidationErrors;
+            } else
+            {
+                // no active step to validate
+                return true;
+            }
+        }
+
+        private static void StepsChangedHandler(DependencyObject obj, DependencyPropertyChangedEventArgs args)
+        {
+            Stepper stepper = (Stepper)obj;
+            stepper.Controller.InitSteps(args.NewValue as IList<IStep>);
+        }
+
+        private void CanExecuteBack(object sender, CanExecuteRoutedEventArgs args)
+        {
+            args.CanExecute = true;
+        }
+
+        private void BackHandler(object sender, ExecutedRoutedEventArgs args)
+        {
+            bool isValid = ValidateActiveStep();
+
+            if (BlockNavigationOnValidationErrors && !isValid)
+            {
+                return;
+            }
+
+            _controller.Back();
+
+            args.Handled = true;
+        }
+
+        private void CanExecuteCancel(object sender, CanExecuteRoutedEventArgs args)
+        {
+            args.CanExecute = true;
+        }
+
+        private void CancelHandler(object sender, ExecutedRoutedEventArgs args)
+        {
+            if (args.Handled)
+            {
+                return;
+            }
+
+            args.Handled = true;
```

Comment:
@CookiesAndCoffee Thank you for your feedback. The commands were intended to control the `Stepper` using navigation buttons. I thought that it is not necessary to let them bubble up. The use case of the cancel button is not eleborate so well.

I just committed a change the let the commands bubble up. Furthermore I added some events to get notified if the `Stepper` starts to navigate to another step. These new events may help you out. 

I hope that the changes will fit your requirements. Let me know if you recognise other bugs, strange behaviour or not pixel-perfect UI elements.